### ISSUE TITLE
The tutorial's revert tests are failing with latest packages  

### ISSUE THREAD

**Issue by klassicd** [ethereum-optimism/plugins#30]
**Describe the bug**
Followed the [tutorial](http://community.optimism.io/tutorial/) with the latest package versions installed.   When I run tests, the ERC 20 passes, and "the basics" from Optimistic ERC20 passes, but the remaining Optimistic tests for "transfer" and "transferFrom" fail. 

**To Reproduce**
1. Clone repo https://github.com/klassicd/optimism-test
2. In optimisim-test folder, run `npm install` 
3. Close optimism-integration and in optimism-integration folder, run `./up.sh` (after installation)
4. In optimisim-test folder, run `npm run test` 

**Expected behavior**
Tests to pass with no issue. 

**Screenshots**
![Screen Shot 2021-03-21 at 2 27 17 PM](https://user-images.githubusercontent.com/18542/111921403-9a430680-8a51-11eb-9bb5-dff98630ac73.png)

If I debug the build of `node_modules/hardhat/internal/hardhat-network/provider/node.js` I see a `VmError { error: 'revert', errorType: 'VmError' }` returned with no `message` property causing the `Cannot read property 'includes' of undefined` error.  which is caused by the compiled code: 

```
if (err === null || err === void 0 ? void 0 : err.message.includes("sender doesn't have enough funds")) {
```

This is compiled from the typescript in `node_modules/hardhat/src/internal/hardhat-network/provider/node.ts`:

```
if (err?.message.includes("sender doesn't have enough funds")) {
```

**System Specs:**
 - OS: Mac 10.14.6
 - Package Version (or commit hash): See fork https://github.com/klassicd/optimism-test

**Additional context**
I think this is related to the error here https://github.com/ethereum-optimism/plugins/issues/25 as the issue is with testing transaction reverts on optimism.

**Comment by klassicd** [comment]
I just tried rewriting the Optimism tests to use the method described here https://github.com/Synthetixio/synthetix/blob/develop/test/optimism/utils/revertOptimism.js

The `VmError { error: 'revert', errorType: 'VmError' }` cannot be parsed by the `getOptimismRevertReason` example.   I'm not exactly what the VmError means.   The optimism-integration is running ok as far as I can tell.


![Screen Shot 2021-03-21 at 2 51 58 PM](https://user-images.githubusercontent.com/18542/111922055-0a9f5700-8a55-11eb-8a9b-15a446716f28.png)

**Comment by smartcontracts** [comment]
Thank you for the report! It seems this is likely the same issue as https://github.com/ethereum-optimism/plugins/issues/17. This is a result of a recent change in hardhat. I opened an issue over there and a fix has been merged, just waiting for a release: https://github.com/nomiclabs/hardhat/issues/1317

Sorry for the inconvenience!

**Comment by smartcontracts** [comment]
Will leave this open for visibility until the fix is merged.

**Comment by klassicd** [comment]
Thanks, can confirm this works with `hardhat@2.1.2-rc.1` and the tutorial's base tests.

**Comment by smartcontracts** [comment]
Thanks for trying that out. I recommend others attempt to update to `hardhat@2.1.2-rc.1` for the moment (but don't forget to bump to a real version once it's released!).

**Comment by fvictorio** [comment]
We just published `hardhat@2.1.2`, that should fix this

**Comment by klassicd** [comment]
Thanks for the great work, verified this works with `hardhat@2.1.2`.
